api/v1alpha1/openapi.yaml

```diff
@@ -2502,6 +2511,254 @@ components:
           type: string
           description: 'Status of the operation. One of: "Success" or "Failure". More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status'
       description: Status is a return value for calls that don't return other objects.
+    DeviceWorkloadStatus:
+      type: object
+      required:
+        - summary
+      properties:
+        applications:
+          type: array
+          description: 'Indicates the current status of applications running on the device.'
+          items:
+            $ref: '#/components/schemas/DeviceApplicationStatus'
+        summary:
+          $ref: '#/components/schemas/DeviceWorkloadSummaryStatus'
+          description: 'Summary status of device workloads.'
+    DeviceApplicationStatus:
+      type: object
+      required:
+        - id
+        - name
+        - status
+      properties:
+        id:
+          type: string
+          description: 'Unique identifier for the application.'
+        name:
+          type: string
+          description: 'Human readable name of the application.'
+        status:
+          $ref: '#/components/schemas/DeviceApplicationStatusType'
+          description: 'Status of the application.'
+        ready:
+          type: string
+          description: 'The number of containers which are ready in the application.'
+        restarts:
+          type: integer
+          description: 'Number of restarts observed for the application.'
+    DeviceWorkloadSummaryStatus:
+      type: object
+      required:
+        - status
+      properties:
+        status:
+          $ref: '#/components/schemas/DeviceWorkloadSummaryType'
+          description: 'Summary status of workloads.'
+        info:
```

Comment:
info provides context to the status. How that is updated is an implementation detail. For example, if you want this to work like a logger an Error status could be updated on sync if additional or updated context exists. If not it could be updated only once on phase/status change.  There are tradeoffs to both but I don't believe the API declares that directly.